crates/mem-pool/src/sync/fan_out.rs

```diff
@@ -0,0 +1,76 @@
+use gw_types::{
+    offchain::DepositInfo,
+    packed::{
+        BlockInfo, L2Transaction, NextMemBlock, RefreshMemBlockMessageUnion, WithdrawalRequest,
+    },
+    prelude::{Builder, Entity, Pack, PackVec},
+};
+use smol::channel::{Receiver, Sender};
+
+use super::mq::{gw_kafka, Produce};
+
+const CHANNEL_BUFFER_SIZE: usize = 500;
+pub(crate) struct FanOutMemBlockActor {
+    receiver: Receiver<RefreshMemBlockMessageUnion>,
+    producer: gw_kafka::Producer,
+}
+
+impl FanOutMemBlockActor {
+    pub(crate) fn new(
+        receiver: Receiver<RefreshMemBlockMessageUnion>,
+        producer: gw_kafka::Producer,
+    ) -> Self {
+        Self { receiver, producer }
+    }
+
+    fn handle(&mut self, msg: RefreshMemBlockMessageUnion) {
+        if let Err(err) = self.producer.send(msg) {
+            log::error!("[Fan out mem block] message failed: {:?}", err);
+        }
+    }
+}
+
+async fn fan_out_handle(mut actor: FanOutMemBlockActor) {
+    log::info!("Fanout handle is started.");
+    while let Ok(msg) = actor.receiver.recv().await {
+        actor.handle(msg);
+    }
+}
+
+pub(crate) struct FanOutMemBlockHandler {
+    sender: Sender<RefreshMemBlockMessageUnion>,
+}
+
+impl FanOutMemBlockHandler {
+    pub(crate) fn new(producer: gw_kafka::Producer) -> Self {
+        let (sender, receiver) = smol::channel::bounded(CHANNEL_BUFFER_SIZE);
+
+        let actor = FanOutMemBlockActor::new(receiver, producer);
+        smol::spawn(fan_out_handle(actor)).detach();
+        Self { sender }
+    }
+
+    pub(crate) fn new_tx(&self, tx: L2Transaction) {
+        let _ = smol::block_on(
+            self.sender
+                .send(RefreshMemBlockMessageUnion::L2Transaction(tx)),
```

Comment:
try_send?

Reply:
Then how can we handle the "channel is full" error if we use try_send here. We cannot discard it. And we cannot block the thread.
I think we have to guarantee the producer will send messages fast enough.

Reply:
There's ```MAX_MEM_BLOCK_TXS```, we can use a slight bigger size for this channel.

Reply:
In case kafka goes down, we can save tx in a VecDeque  in ```FanOutMemBlockActor```.

Reply:
We cannot do this. Kafka cluster has higher Availability than godwoken, apparently. It has quorum that we don't. We should not backup data for it.
Instead, kafka should be monitored very well.

Reply:
I mean just save unsend txs if channel is full, not backup all txs. And try send them first.